src/sortStudents.ts

```diff
@@ -1,16 +1,58 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function findAverage(grades: number[]): number {
+    return grades
+      .reduce((sum, currentGrade) => sum + currentGrade) / grades.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+      return order === 'asc'
+        ? copyStudents.sort((a, b) => a.name.localeCompare(b.name))
+        : copyStudents.sort((a, b) => b.name.localeCompare(a.name));
+    case SortType.Surname:
+      return order === 'asc'
+        ? copyStudents.sort((a, b) => a.surname.localeCompare(b.surname))
+        : copyStudents.sort((a, b) => b.surname.localeCompare(a.surname));
```

Comment:
Checklist. Avoid duplication in these cases.
```suggestion
  const sortOrder = order === 'asc' ? 1 : -1;
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
      return sortOrder * copyStudents.sort((a, b) => a[sortBy].localeCompare(b[sortBy]))
```

Reply:
Same for the other cases